sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/GroupByKey.java

```diff
@@ -162,6 +164,44 @@ public static void applicableTo(PCollection<?> input) {
       throw new IllegalStateException(
           "GroupByKey must have a valid Window merge function.  " + "Invalid because: " + cause);
     }
+
+    // Validate that the trigger does not finish before garbage collection time
+    if (!triggerIsSafe(windowingStrategy)) {
+      throw new IllegalArgumentException(
+          String.format(
+              "Unsafe trigger may finish and drop remaining data: %s",
```

Comment:
This would probably rise many questions on mailing list, can we add here a link to a document describing what just happened?

Reply:
Interesting idea. Yes, I can make a world-readable doc that is a short summary of the issue and why this is disabled and how to contact us for those rare cases where someone wants it back. We could give them a pipeline option. But I would wait until someone asks for it, to avoid implementing something dangerous.